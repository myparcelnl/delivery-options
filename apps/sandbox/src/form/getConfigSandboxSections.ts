import {CarrierSetting, KEY_CONFIG, OptionGroup} from '@myparcel-do/shared';
import {type SandboxOptionGroup, type SettingsSection} from '../types';
import {resolveSandboxSection} from './resolveSandboxSection';

const CONFIG_GROUPS = Object.freeze([
  {
    name: OptionGroup.Delivery,
    items: [CarrierSetting.AllowDeliveryOptions],
    children: [
      {
        name: OptionGroup.DeliveryMoments,
        items: [
          CarrierSetting.AllowStandardDelivery,
          CarrierSetting.AllowMorningDelivery,
          CarrierSetting.AllowEveningDelivery,
          CarrierSetting.AllowSameDayDelivery,
          CarrierSetting.AllowMondayDelivery,
          CarrierSetting.AllowSaturdayDelivery,
        ],
      },
      {
        name: OptionGroup.ShipmentOptions,
        items: [CarrierSetting.AllowOnlyRecipient, CarrierSetting.AllowSignature],
      },
      {
        name: OptionGroup.PackageTypes,
        items: [CarrierSetting.PricePackageTypeMailbox, CarrierSetting.PricePackageTypeDigitalStamp],
      },
    ],
  },

  {
    name: OptionGroup.Pickup,
    items: [CarrierSetting.AllowPickupLocations],
  },
] satisfies SandboxOptionGroup[]);

export const getConfigSandboxSections = (prefix?: string): SettingsSection[] => {
  return CONFIG_GROUPS.map((group) => resolveSandboxSection(group, KEY_CONFIG + (prefix ? `.${prefix}` : '')));
};
