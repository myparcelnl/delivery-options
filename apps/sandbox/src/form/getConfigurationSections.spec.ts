import {beforeEach, describe, expect, it} from 'vitest';
import {createPinia, setActivePinia} from 'pinia';
import {getConfigSandboxSections} from './getConfigSandboxSections';

describe('getConfigurationSections', () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it('gets sandbox option groups', () => {
    const sections = getConfigSandboxSections();

    expect(sections).toMatchSnapshot();
  });
});
