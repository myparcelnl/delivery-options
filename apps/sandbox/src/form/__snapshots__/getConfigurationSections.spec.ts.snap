// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getConfigurationSections > gets sandbox option groups 1`] = `
[
  {
    "fields": [
      {
        "description": "delivery",
        "fields": [
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowDeliveryOptions",
              "name": "config.allowDeliveryOptions",
              "props": {
                "parents": [],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceStandardDelivery",
              "name": "config.priceStandardDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowMorningDelivery",
              "name": "config.allowMorningDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceMorningDelivery",
              "name": "config.priceMorningDelivery",
              "props": {
                "parents": [
                  "allowMorningDelivery",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowEveningDelivery",
              "name": "config.allowEveningDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceEveningDelivery",
              "name": "config.priceEveningDelivery",
              "props": {
                "parents": [
                  "allowEveningDelivery",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowSameDayDelivery",
              "name": "config.allowSameDayDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceSameDayDelivery",
              "name": "config.priceSameDayDelivery",
              "props": {
                "parents": [
                  "allowSameDayDelivery",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowMondayDelivery",
              "name": "config.allowMondayDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceMondayDelivery",
              "name": "config.priceMondayDelivery",
              "props": {
                "parents": [
                  "allowMondayDelivery",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowSaturdayDelivery",
              "name": "config.allowSaturdayDelivery",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceSaturdayDelivery",
              "name": "config.priceSaturdayDelivery",
              "props": {
                "parents": [
                  "allowSaturdayDelivery",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowOnlyRecipient",
              "name": "config.allowOnlyRecipient",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceOnlyRecipient",
              "name": "config.priceOnlyRecipient",
              "props": {
                "parents": [
                  "allowOnlyRecipient",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowSignature",
              "name": "config.allowSignature",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "priceSignature",
              "name": "config.priceSignature",
              "props": {
                "parents": [
                  "allowSignature",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowPackageTypeDigitalStamp",
              "name": "config.allowPackageTypeDigitalStamp",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "pricePackageTypeDigitalStamp",
              "name": "config.pricePackageTypeDigitalStamp",
              "props": {
                "parents": [
                  "allowPackageTypeDigitalStamp",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowPackageTypeMailbox",
              "name": "config.allowPackageTypeMailbox",
              "props": {
                "parents": [
                  "allowDeliveryOptions",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "pricePackageTypeMailbox",
              "name": "config.pricePackageTypeMailbox",
              "props": {
                "parents": [
                  "allowPackageTypeMailbox",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
        ],
        "key": "delivery",
      },
    ],
    "label": "delivery",
  },
  {
    "fields": [
      {
        "description": "pickup",
        "fields": [
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormToggleInput.vue",
                "__name": "FormToggleInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "key": "config",
              "label": "allowPickupLocations",
              "name": "config.allowPickupLocations",
              "props": {
                "parents": [],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
          {
            "Component": {
              "render": [Function],
              "setup": [Function],
            },
            "field": {
              "component": {
                "__file": "/Users/edie/Sites/myparcelnl/delivery-options/apps/sandbox/src/components/form/input/FormCurrencyInput.vue",
                "__name": "FormCurrencyInput",
                "emits": [
                  "update:modelValue",
                ],
                "props": {
                  "element": {
                    "required": true,
                    "type": null,
                  },
                  "modelValue": {
                    "required": true,
                    "type": null,
                  },
                },
                "render": [Function],
                "setup": [Function],
              },
              "disabledWhen": [Function],
              "key": "config",
              "label": "pricePickup",
              "name": "config.pricePickup",
              "props": {
                "parents": [
                  "allowPickupLocations",
                ],
              },
              "ref": undefined,
              "validators": [],
              "visibleWhen": [Function],
            },
            "ref": undefined,
          },
        ],
        "key": "pickup",
      },
    ],
    "label": "pickup",
  },
]
`;
